test/RelayHubGas.test.ts

```diff
@@ -0,0 +1,699 @@
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { BigNumber, ContractReceipt, Wallet, providers } from 'ethers';
+import {
+  UtilToken,
+  SmartWalletFactory,
+  Penalizer,
+  RelayHub,
+  SmartWallet,
+} from '@rsksmart/rif-relay-contracts';
+import {
+  CommonEnvelopingRequestBody,
+  ESTIMATED_GAS_CORRECTION_FACTOR,
+  EnvelopingRequest,
+  EnvelopingRequestData,
+  INTERNAL_TRANSACTION_ESTIMATED_CORRECTION,
+  RelayRequest,
+  RelayRequestBody,
+} from '@rsksmart/rif-relay-client';
+import {
+  TestRecipient,
+  TestVerifierEverythingAccepted,
+} from '../typechain-types';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import {
+  getSuffixDataAndSignature,
+  createSmartWalletFactory,
+  createSupportedSmartWallet,
+  RSK_URL,
+} from './utils/TestUtils';
+
+function logGasOverhead(gasOverhead: BigNumber) {
+  const bgMagenta = '\x1B[45m';
+  const fgWhite = '\x1B[37m';
+  const reset = '\x1b[0m';
+  console.log(
+    bgMagenta,
+    fgWhite,
+    `Enveloping Overhead Gas: ${gasOverhead.toString()}`,
+    reset
+  );
+}
+
+const deployContract = <Contract>(contract: string) => {
+  return ethers
+    .getContractFactory(contract)
+    .then((contractFactory) => contractFactory.deploy() as Contract);
+};
+
+describe('RelayHub GasEstimation', function () {
+  let penalizer: Penalizer;
+  let relayHub: RelayHub;
+  let verifier: TestVerifierEverythingAccepted;
+  let recipient: TestRecipient;
+  let token: UtilToken;
+  let factory: SmartWalletFactory;
+  let relayWorker: SignerWithAddress;
+  let relayManager: SignerWithAddress;
+  let relayOwner: SignerWithAddress;
+  let fundedAccount: SignerWithAddress;
+  let relayHubSigner: SignerWithAddress;
+  let owner: Wallet;
+  let provider: providers.JsonRpcProvider;
+  const gasPrice = 1;
+  const gasLimit = 4e6;
+
+  const deployHub = (penalizerAddress: string) => {
```

Comment:
Couldn't we use the same function defined in `RelayHub.test.ts`?

Reply:
even better, the deployRelayHub function in testUtils is now working so I'm using that one.